formats/html.ts

```diff
@@ -21,24 +73,340 @@ export class HtmlImporter extends FormatImporter {
 			return;
 		}
 
-		let results: ImportResult = {
-			total: 0,
+		const results = await Promise.all(files
+			.map(file => this.processFile(folder, file)));
+		this.showResult({
+			total: results.map(({ total }) => total).reduce((left, right) => left + right, 0),
+			failed: results.map(({ failed }) => failed).flat(),
+			skipped: results.map(({ skipped }) => skipped).flat()
+		});
+	}
+
+	async processFile(folder: TFolder, file: PickedFile) {
+		const results: ImportResult = {
+			total: 1,
+			failed: [],
 			skipped: [],
-			failed: []
 		};
+		if (!(file instanceof NodePickedFile)) {
+			results.skipped.push(file.name);
+			return results;
+		}
+		let mdFile: TFile | null = null;
+		try {
+			let mdContent = htmlToMarkdown(await file.readText());
+			const pathURL = nodeUrl.pathToFileURL(file.filepath);
+			mdFile = await this.saveAsMarkdownFile(folder, file.basename, "");
+
+			const ast = [];
```

Comment:
I also worked on translating wikilinks in the [Notion importer](https://github.com/joshuatazrein/obsidian-importer/blob/929590bd13f19490c028ef37e0b401f5f0f705ed/formats/notion/convert-to-md.ts#L191).

Reply:
> Instead of rolling our own parser here, perhaps it would be easier to work on the original HTML? What I suggest doing is to use DOMParser to parse the HTML into a DOM tree first, then querySelector on the `<img>` elements and working off them. [(See how this is implemented in the Notion importer)](https://github.com/obsidianmd/obsidian-importer/pull/34/files#diff-dcb7a54e5cb2f4c3a888c6eb457ebfc648ec9534b52beeb4202fea10e9514e30R85)

Done. `<img>`, `<audio>`, and `<video>` elements are now queried.

> It unfortunately won't help us generate wikilinks instead of markdown links, but one possibility is to go back to match the URL in the final markdown text and attempt to regex the before/after from the position where the URL was matched.

The Markdown parsing is now done using the builtin parser via `MetadataCache` (by first fixing the embed links outputted by `htmlToMarkdown`), so regex is not even needed to find the embed links to replace. The wikilink generation is already handled by `generateMarkdownLink`.